### PR TITLE
drop end of life pythons (<3.7)

### PR DESCRIPTION
**Pull request by benbariteau**
Most of the changes here are done by `pyupgrade --py37-plus`. Notable things that aren't:
- Migrating to mirrors-autopep8 pre-commit hook from autopep8-wrapper
- Adding `python_requires='>=3.7'`
- Changing test setup to run under python 3.7 and 3.8. This required upgrading the docker image and making some other changes like adding deadsnakes.
- Removing installation of `tox-pip-extensions` within github workflow. This never made sense as public pypi can't use pip-custom-platform and venv-update doesn't help in a CI environment.

### CODE REVIEW

**Review by JayH5 — DISMISSED** [review]
Only so much diff I can review but LGTM, assuming the auto-formatting tools are doing their job correctly 👍

File: kafka_utils/kafka_check/metadata_file.py [hunk 1]
```diff
@@ -13,10 +13,10 @@ def _read_generated_broker_id(meta_properties_path):
...
     try:
-        with open(meta_properties_path, 'r') as f:
+        with open(meta_properties_path) as f:
             broker_id = _parse_meta_properties_file(f)
-    except IOError:
-        raise IOError(
+    except OSError:
+        raise OSError(
             "Cannot open meta.properties file: {path}"
             .format(path=meta_properties_path),
```
[the remark this answers is not in the document]
Comment (benbariteau) [reply kafka_utils/kafka_check/metadata_file.py:21]: I guess the linebreak isn't detected by pyupgrade. Up to you if you want me to fix it.

File: kafka_utils/kafka_check/main.py [hunk 2]
```diff
@@ -149,7 +145,7 @@ def validate_args(args):
             except Exception as e:
                 terminate(
                     status_code.WARNING,
-                    prepare_terminate_message("{}".format(e)),
+                    prepare_terminate_message(f"{e}"),
```
[the remark this answers is not in the document]
Comment (benbariteau) [reply kafka_utils/kafka_check/main.py:148]: This is equivalent.

[the remark this answers is not in the document]
Comment (JayH5) [reply kafka_utils/kafka_check/main.py:148]: I know, I've just never seen converting to a string like that.

**Review by JayH5 — APPROVED** [review]

**Pull request merged**